### PR TITLE
[WIP] Add appveyor.xml to build on Windows.

### PR DESCRIPTION
**Pull request by sheredom**
I'm testing Appveyor integration (for Windows builds).

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+version: '{build}'
+
+skip_tags: true
+
+install: []
+
+environment:
+  matrix:
+    - VSVERSION: Visual Studio 12
```
Comment (dneto0) [inline appveyor.yml]: I'd prefer to stay away from older MSVC compilers.  Their C++11 support is sometimes lacking.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - Win32
+  - x64
+
+configuration:
+  - Debug
+  - Release
+
+build_script:
+  - git clone https://github.com/google/googletest.git external/googletest
+  - md build
```
Comment (dneto0) [inline appveyor.yml]: Take a look at https://github.com/google/shaderc/blob/master/.appveyor.yml 
It uses cmake --build, which potentially frees up a dependency.  I also find it easier to read. :-)
Also it tests via ctest.  That will let us add other test executables without having to update this config.

File: source/util/bitutils.h [hunk 3]
```diff
@@ -27,6 +27,8 @@
 #ifndef LIBSPIRV_UTIL_BITUTILS_H_
 #define LIBSPIRV_UTIL_BITUTILS_H_
 
+#include <stdint.h>
```
Comment (dneto0) [inline source/util/bitutils.h]: I guess this is a symptom of VS 12?
In C++ code like this we'd prefer #include <cstdint>

**Comment by dneto0** [comment]
See https://github.com/google/shaderc/blob/master/.appveyor.yml for another AppVeyor config that we've been using.  I prefer the use of cmake to execute the build, and ctest to run the tests.

**Comment by sheredom** [comment]
Tbh now that I've seen the shaderc one? I'd be totally happy bringing something like that over. I just wanted some form of Windows testing in the project! :)

File: appveyor.yml (hunk 1 above)
Comment (sheredom) [inline appveyor.yml]: This is the VS I personally use - 2015 is far too temperamental for my liking!

File: source/util/bitutils.h (hunk 3 above)
Comment (sheredom) [inline source/util/bitutils.h]: Yup, VS12 issue.

**Comment by antiagainst** [comment]
@sheredom This is kinda tested under windows because it is pulled in as a dependency for shaderc. But, yeah, definitely worth to do that in its own repo.

But AFAIK, there are some permission issue with appveyor (it requires to access private repo, if I recall that correctly) which may need to be allowed by the org (KhronosGroup) first.

**Comment by sheredom** [comment]
Righto, I'm gonna reject my own MR then! :)

**Pull request closed without merging**